Implementacion de CRUD para providers

Werfran opened the pull request:
# Description

Se agregaron los métodos CRUD para proveedores

# Issues

# Opinion

No estoy seguro si el filtro para el seach deberia usar el metodo contains, pero lo use ya que es lo que se uso para el Store

Werfran committed:
Sibuda, se crearon las clases del enunciado, falta ver que le pasa que no devuelve nada los gets

Werfran committed:
Se completaron los gets y el post pasando las pruebas del provider,

Werfran committed:
Pasados todos los test

Werfran committed:
se arreglo el serach de providers

gerardoaquino25 commented on Stock.AppService/Services/ProviderService.cs:
```diff
@@ -0,0 +1,61 @@
+
+        public new Provider Create(Provider provider)
+        {
+            //if(NombreUnico(provider.Name) && EmailUnico(provider.Email) && TelefonoUnico(provider.Phone))
```
Al quedar comentado, no esta realizando estas validaciones. No es apropiado que no se realice algún tipo de validación al respecto.

gerardoaquino25 commented on Stock.Api/Controllers/ProviderController.cs:
```diff
@@ -0,0 +1,143 @@
+        }
+
+        [HttpPut("{id}")]
+        public void Put(string id, [FromBody] ProviderDTO value)
```
Que pasa si el parámetro id es distinto al del id contenido dentro del parámetro value?
Como se podría solucionar ese problema?

gerardoaquino25 commented on Stock.Api/Controllers/ProviderController.cs:
```diff
@@ -0,0 +1,143 @@
+        }
+
+        [HttpPost]
+        public ActionResult Post([FromBody] ProviderDTO value)
```
Cual es la diferencia entre usar como respuesta un ActionResult, ActionResult\<ProviderDTO\> y ProviderDTO? Cual te parece más adecuado?
Te dejo links interesantes al respecto:
https://docs.microsoft.com/en-us/aspnet/core/web-api/action-return-types?view=aspnetcore-6.0
https://swagger.io/specification/

gerardoaquino25 commented on Stock.Api/Controllers/ProviderController.cs:
```diff
@@ -0,0 +1,143 @@
+            catch (Exception ex)
+            {
+                logger.LogCritical(ex.StackTrace);
+                return Ok(new { Success = false, Message = "The id is already in use" });
```
Te parece correcto arrojar el mensaje que esta devolviendo en caso de excepción?
Que tipo de Http status code estas enviando en este caso? Te parece correcto?
Links:
https://restfulapi.net/http-status-codes/

Werfran committed:
-Se creo NameService para verificar la existencia de un nombre repetido para provider y store.
-Se arreglaron las observaciones hechas por los tutores.
-Ahora en el metodo put tiene que coincidir el id del provider del body con el de la URL.
-Se cambio a donde apunta el front para coincidir con la URL del back.
-Hay problema de cors, se agrego codigo pero se comento en el Startup.cs tratando de arreglarlo, no hubo caso.
